### PR TITLE
Fix electrostatic solver

### PR DESCRIPTION
**Pull request by RemiLehe**
# Overview

This PR is a major refactoring of the electrostatic solver: when the user specifies `do_electrostatic=1`, instead of using the old `EvolveES` (which was badly broken), this PR now uses the regular `EvolveEM` (which was renamed as ~~`EvolveE`~~ `Evolve`) with small changes in the main loop in order to accomodate the electrostatic solver instead of the electromagnetic solver.

# List of the main changes

- In the main PIC loop: if `do_electrostatic`, then the electrostatic field is computed at the beginning of the PIC loop, and the Maxwell solver is not called anymore at the end of the PIC loop.
- In the particle's `Evolve` function (which gathers, push, and deposit the particles): if `do_electrostatic`, then the current is not deposited anymore ; the charge is also not deposited at t=n+1 because it could lead to out-of-bounds - due to the potentially large timestep in the ES solver.
- In order to allow the above, `do_electrostatic` is now a static, public variable of `WarpX` (allows easy access from outside of the `WarpX` class)

# Implications and side notes

- As a consequence of this PR, a number of functions in WarpX have become obsolete and unused (all the functions that were called by `EvolveES`). These functions will be removed in a separate PR.
- Note that running in electrostatic mode does *not* require to compile the code with `DO_ELECTROSTATIC` anymore ; this is because the electrostatic solver is already compiled in EM mode (as it is sometimes used for initialization)
- The main PIC loop is currently not very efficiently, since e.g. it is still calling `SynchronizeCurrent` for the electrostatic mode. This (and similar optimizations) will be done in a separate PR.

# Tests

With these changes, the electrostatic test runs and produces reasonable-looking results. Quantitative agreement is currently being investigated.

### CODE REVIEW

File: Source/Evolve/WarpXEvolve.cpp [hunk 1]
```diff
@@ -375,50 +386,51 @@ WarpX::OneStep_nosub (Real cur_time)
     if (do_pml && pml_has_particles) CopyJPML();
     if (do_pml && do_pml_j_damping) DampJPML();
 
+    if (!do_electrostatic) {
+    // Electromagnetic solver:
     // Push E and B from {n} to {n+1}
     // (And update guard cells immediately afterwards)
 #ifdef WARPX_USE_PSATD
```
Comment (RemiLehe) [inline Source/Evolve/WarpXEvolve.cpp]: The only changes below are the re-indentation fo the whole code block. 
Unfortunately, the Github interface does not show this very well.

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -1116,21 +1117,6 @@ WarpX::RefRatio (int lev)
     return GetInstance().refRatio(lev);
 }
```
Comment (RemiLehe) [inline Source/WarpX.cpp]: This is not needed anymore since:
- `EvolveEM` has been renamed as `EvolveE`
- `EvolveES` is not used anymore.

Comment (MaxThevenet) [reply Source/WarpX.cpp]: > EvolveEM has been renamed as EvolveE

do you mean
> EvolveEM has been renamed as Evolve

?

**Comment by RemiLehe** [comment]
Thanks for reviewing!

- I just added the documentation for `do_electrostatic` ; thanks for the reminder :)
- You are absolutely right: the description should mention `Evolve`, not `EvolveE` ; I edited it.
- Yes, there will be a following PR that removes this compilation flag and the associated code.

**Pull request merged**